### ISSUE TITLE
Adapt the indoor routing algorithm to integrate seamlessly with MapLibre GL

### ISSUE THREAD

**Issue by KnotzerIO** [openindoormap/openindoormaps#30]

**Comment by Yunhh-Hahn** [comment]
Have u guys looked into osrm cause maplibre GL direction is using it but because osrm hosted api is driving, it cannot do walking

**Comment by KnotzerIO** [comment]
Yes, of course. We also use it for outdoor route visualization, but for indoor routes, the route calculation doesn't have to be done on the server; instead, it can be done on the client. That's what we are trying to achieve.

**Comment by Yunhh-Hahn** [comment]
does this mean u are using osrm for outdoor route and for indoor route it's purely on the client or u are aiming openIndoormaps to be an entirely client side application

**Comment by KnotzerIO** [comment]
only the indoor route calculation should be done on the client

**Comment by Yunhh-Hahn** [comment]
any reason why do u want to make it only client for indoor ? It's usually bad practice to reinvent the wheel especially considering how powerful routing engine like osrm and vallahala exists with a very big community already. Furthermore, it will complicate the routing process if the user want to navigate something like within their uni campus as it can involves going through multiple indoor and outdoor spaces.

Also do u guy got any discord or something, I still has a lots of question i.e I am gonna bloat the comment of this issue... and I am very interested in contributing

**Comment by KnotzerIO** [comment]
I understand your point, but we decided to use client-side routing because it would be more performant to do the route calculation on the client since the graph is very small. The functionality of other routing services would be overkill for simple indoor routing. Other solutions also handle indoor routing on the client side.

Very nice of you that you want to join the team, we do have a discord channel. we just need to make a few changes,and then we will invite you so we can talk

**Comment by KnotzerIO** [comment]
Here you can join us: https://discord.gg/gaDkUGag

**Comment by KnotzerIO** [comment]
I amclosing this issue as indoor routing is now working, though not yet perfected. The implementation is similar to the MapLibre GL direction.
